client/trino-jdbc/src/main/java/io/trino/jdbc/Row.java

```diff
@@ -34,9 +35,12 @@
 
     public List<RowField> getFields()
     {
-        return row.getFields().stream()
-                .map(RowField::new)
-                .collect(toImmutableList());
+        List<io.trino.client.RowField> fields = row.getFields();
```

Comment:
Did you benchmark this? Is this really faster?

Reply:
You can always use own collector definition with a builder having expected size:

```
    static <E> Collector<E, ?, ImmutableList<E>> toImmutableListWithExpectedSize(int expectedSize)
    {
        return (Collector) Collector.of(
                () -> ImmutableList.builderWithExpectedSize(expectedSize),
                ImmutableList.Builder::add,
                (builder, otherBuilder) -> builder.addAll(otherBuilder.build()),
                ImmutableList.Builder::build);
    }
```